hack/make/.binary

```diff
@@ -62,11 +62,19 @@ source "${MAKEDIR}/.go-autogen"
 		fi
 	fi
 
+	# This is a workaround to have buildinfo with deps embedded in the binary. We
+	# need to create a go.mod file before building with -modfile=vendor.mod,
+	# otherwise it fails with: "-modfile cannot be used to set the module root directory."
+	if [ ! -f "go.mod" ]; then
+		printf '%s\n\n%s' 'module github.com/docker/docker' 'go 1.19' > "go.mod"
+		trap 'rm -f go.mod' EXIT
+	fi
+
 	echo "Building $([ "$DOCKER_STATIC" = "1" ] && echo "static" || echo "dynamic") $DEST/$BINARY_FULLNAME ($PLATFORM_NAME)..."
 	if [ -n "$DOCKER_DEBUG" ]; then
 		set -x
 	fi
-	go build -o "$DEST/$BINARY_FULLNAME" "${BUILDFLAGS[@]}" -ldflags "$LDFLAGS $LDFLAGS_STATIC $DOCKER_LDFLAGS" ${GO_PACKAGE}
+	GO111MODULE=on go build -mod=vendor -modfile=vendor.mod -o "$DEST/$BINARY_FULLNAME" "${BUILDFLAGS[@]}" -ldflags "$LDFLAGS $LDFLAGS_STATIC $DOCKER_LDFLAGS" ${GO_PACKAGE}
```

Comment:
I'm late here, but was there a reason not to prefer the `with-go-mod` wrapper script we already have?

Reply:
Log output was a bit clunky using `with-go-mod` and also the dependency on `git` was an issue as we ignore `.git` in `.dockerignore`: https://github.com/moby/moby/pull/44917/files#diff-fa876a8e859e8583907d556eeed2e4b8cc98d7847b2124521aaef1e9984f50cf